.github/workflows/check-cuda-formatted.yml

```diff
@@ -0,0 +1,36 @@
+# This file was copied mostly from check-maintainers-sorted.yaml
+name: Check that CUDA files are formatted
+
+on:
+  pull_request_target:
+    paths:
+      - pkgs/development/cuda-modules/**.nix
+      - pkgs/test/cuda/**.nix
+      - pkgs/top-level/cuda-packages.nix
+permissions:
+  contents: read
+
+jobs:
+  nixos:
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'NixOS'
+    steps:
+      - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+        with:
+          # pull_request_target checks out the base branch by default
+          ref: refs/pull/${{ github.event.pull_request.number }}/merge
+      - uses: cachix/install-nix-action@8887e596b4ee1134dae06b98d573bd674693f47c # v26
+        with:
+          # explicitly enable sandbox
+          extra_nix_config: sandbox = true
+      - name: Check that CUDA files are formatted
+        run: |
+          nix-shell \
+            --pure \
+            --packages nixfmt-rfc-style \
+            --run \
+              'nixfmt \
+                --check \
+                pkgs/development/cuda-modules \
+                pkgs/test/cuda \
+                pkgs/top-level/cuda-packages.nix'
```

Comment:
In line with making the check more generic, I wonder if we should replace this with the list of files that changed between `HEAD` and the merge-base. This way we only "complain" about files brought into scope by PR authors

Reply:
That would be neat, but I don't want to dig too far into the GitHub actions stuff. I firmly believe there should be a community-maintained GitHub action which implements that requirement -- it's super common to only want to run against stuff that has changed! I don't know I've seen that done outside pre-commit hooks, but I haven't look too closely.

Reply:
I was told the same when I wrote the nix-parse workflow, and I didn't want to interact with GH actions either… anyway, this is the solution I was hinted to: https://github.com/NixOS/nixpkgs/blob/master/.github/workflows/nix-parse.yml